File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderAssessmentResourceTest.java
package uk.gov.justice.hmpps.prison.api.resource.impl;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;
import org.springframework.web.client.HttpClientErrorException;
import uk.gov.justice.hmpps.prison.api.resource.OffenderAssessmentResource;
import uk.gov.justice.hmpps.prison.api.support.CategoryInformationType;
import uk.gov.justice.hmpps.prison.service.InmateService;

import static org.assertj.core.api.Assertions.assertThatThrownBy;
import static org.mockito.Mockito.verify;

@ExtendWith(MockitoExtension.class)
public class OffenderAssessmentResourceTest {
    @Mock
    private InmateService inmateService;
    private OffenderAssessmentResource offenderAssessmentResource;

    @BeforeEach
    public void setUp() {
        offenderAssessmentResource = new OffenderAssessmentResource(inmateService, null);
    }

    @Test
    public void getOffenderCategorisationsInvalidType() {
        assertThatThrownBy(() -> offenderAssessmentResource.getOffenderCategorisations("LEI", "INVALID_CAT_TYPE", null))
                .isInstanceOf(HttpClientErrorException.class);
    }

    @Test()
    public void getOffenderCategorisations() {
        offenderAssessmentResource.getOffenderCategorisations("LEI", CategoryInformationType.CATEGORISED.name(), null);
        verify(inmateService).getCategory("LEI", CategoryInformationType.CATEGORISED, null);
    }
}

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderAssessmentResourceTest.kt
package uk.gov.justice.hmpps.prison.api.resource.impl

import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test
import org.mockito.Mockito.verify
import org.mockito.kotlin.mock
import org.springframework.web.client.HttpClientErrorException
import uk.gov.justice.hmpps.prison.api.resource.OffenderAssessmentResource
import uk.gov.justice.hmpps.prison.api.support.CategoryInformationType.CATEGORISED
import uk.gov.justice.hmpps.prison.service.InmateService

class OffenderAssessmentResourceTest {
  private val inmateService = mock<InmateService>()
  private val offenderAssessmentResource = OffenderAssessmentResource(inmateService, null)

  @Test
  fun offenderCategorisationsInvalidType() {
    assertThatThrownBy { offenderAssessmentResource.getOffenderCategorisations("LEI", "INVALID_CAT_TYPE", null) }
      .isInstanceOf(HttpClientErrorException::class.java)
  }

  @Test
  fun offenderCategorisations() {
    offenderAssessmentResource.getOffenderCategorisations("LEI", CATEGORISED.name, null)
    verify(inmateService).getCategory("LEI", CATEGORISED, null)
  }
}
